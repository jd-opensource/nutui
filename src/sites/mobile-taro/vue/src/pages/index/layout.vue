<template>
  <div class="index">
    <Header :compName="compName" />
    <iframe :src="src"></iframe>
  </div>
</template>

<script lang="ts">
import { getCurrentInstance } from '@tarojs/taro';
import { ref, computed, Ref } from 'vue';
import Header from '../../components/header.vue';
export default {
  name: 'NutUI',
  components: {
    Header
  },
  setup() {
    let compName = ref('') as Ref;

    let $instance = getCurrentInstance();

    compName.value = $instance.router?.params.n;

    const src = computed(() => {
      return `/#/${$instance.router?.params.e}/pages/${$instance.router?.params.n}/index`;
    });

    return {
      compName,
      src
    };
  }
};
</script>
<style lang="scss">
.index {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow-y: auto;
}
iframe {
  border: 0;
  position: fixed;
  top: 57px;
  bottom: 0;
  left: 0;
  width: 100%;
  height: calc(100% - 57px);
}
</style>
