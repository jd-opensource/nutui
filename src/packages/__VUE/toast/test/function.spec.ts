import { showToast } from '../index';

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

describe('function toast', () => {
  test('show loading toast', async () => {
    showToast.loading('加载中', {
      duration: 50
    });
    expect(document.querySelector('.nut-toast.nut-toast-loading')).toBeTruthy();
    const text = document.querySelector('.nut-toast .nut-toast-text') as HTMLDivElement;
    expect(text.innerHTML).toEqual('加载中');
    await sleep(50);
    const loading = document.querySelector('.nut-toast.nut-toast-loading');
    expect(loading).toBeNull();
  });

  test('show text toast', async () => {
    showToast.text('文案', {
      duration: 50,
      title: '标题文字'
    });
    await sleep();
    const textToast = document.querySelector('.nut-toast.nut-toast-center') as HTMLElement;
    expect(textToast.innerHTML).toContain('标题文字');
    await sleep(100);
    expect(textToast.style.display).toEqual('none');
  });
  test('show fail toast', async () => {
    showToast.fail('文案', {
      duration: 0
    });
    await sleep();
    const failToast = document.querySelector('.nut-toast');
    expect(failToast).toMatchSnapshot();
  });
  test('show warn toast', async () => {
    showToast.warn('文案', {
      duration: 0
    });
    await sleep();
    const failToast = document.querySelector('.nut-toast');
    expect(failToast).toMatchSnapshot();
  });
});
