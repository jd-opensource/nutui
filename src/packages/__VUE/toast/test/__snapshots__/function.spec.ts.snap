// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`function toast > show fail toast 1`] = `
<view
  class="nut-toast nut-toast-center nut-toast-has-icon nut-toast-base toast-fade-enter-from toast-fade-enter-active"
  iconsize="20"
  style=""
>
  <view
    class="nut-toast-inner"
    style="text-align: center;"
  >
    <view
      class="nut-toast-icon-wrapper"
    >
      <svg
        aria-labelledby=""
        class="nut-icon"
        role="presentation"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 1024C229.831 1024 0 794.169 0 512S229.831 0 512 0s512 229.831 512 512-229.831 512-512 512zm0-955.733C266.24 68.267 68.267 266.24 68.267 512S266.24 955.733 512 955.733 955.733 757.76 955.733 512 757.76 68.267 512 68.267zm111.502 282.169c13.654-13.654 34.134-13.654 47.787 0 11.378 11.377 13.653 31.857 2.275 45.51l-2.275 2.276-111.502 111.502 111.502 111.503c13.653 13.653 13.653 34.133 0 47.786-13.653 13.654-34.133 13.654-47.787 0L512 557.511 400.498 669.013c-11.378 11.378-31.858 13.654-45.511 2.276l-2.276-2.276c-11.378-11.377-13.653-31.857-2.275-45.51l2.275-2.276 111.502-111.503-111.502-111.502c-13.653-13.653-13.653-34.133 0-47.786s34.133-13.654 47.787 0L512 461.938l111.502-111.502z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </view>
    <!--v-if-->
    <view
      class="nut-toast-text"
    >
      文案
    </view>
  </view>
</view>
`;

exports[`function toast > show warn toast 1`] = `
<view
  class="nut-toast nut-toast-center nut-toast-has-icon nut-toast-base toast-fade-enter-from toast-fade-enter-active"
  iconsize="20"
  style=""
>
  <view
    class="nut-toast-inner"
    style="text-align: center;"
  >
    <view
      class="nut-toast-icon-wrapper"
    >
      <svg
        aria-labelledby=""
        class="nut-icon"
        role="presentation"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 1024C229.831 1024 0 794.169 0 512S229.831 0 512 0s512 229.831 512 512-229.831 512-512 512zm0-955.733C266.24 68.267 68.267 266.24 68.267 512S266.24 955.733 512 955.733 955.733 757.76 955.733 512 757.76 68.267 512 68.267zm111.502 282.169c13.654-13.654 34.134-13.654 47.787 0 11.378 11.377 13.653 31.857 2.275 45.51l-2.275 2.276-111.502 111.502 111.502 111.503c13.653 13.653 13.653 34.133 0 47.786-13.653 13.654-34.133 13.654-47.787 0L512 557.511 400.498 669.013c-11.378 11.378-31.858 13.654-45.511 2.276l-2.276-2.276c-11.378-11.377-13.653-31.857-2.275-45.51l2.275-2.276 111.502-111.503-111.502-111.502c-13.653-13.653-13.653-34.133 0-47.786s34.133-13.654 47.787 0L512 461.938l111.502-111.502z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </view>
    <!--v-if-->
    <view
      class="nut-toast-text"
    >
      文案
    </view>
  </view>
</view>
`;
